### PR TITLE
Fix Bug 1137061 - Firefox Release Notes list shows unsorted sub-versions

### PR DESCRIPTION
**Pull request by kyoshino**
* Sort all minor version numbers properly.
* Solved duplicated 33.1.1.

### CODE REVIEW

File: bedrock/releasenotes/views.py [hunk 1]
```diff
@@ -173,15 +176,17 @@ def releases_index(request, product):
         # The version numbering scheme of Firefox changes sometimes. The second
         # number has not been used since Firefox 4, then reintroduced with
         # Firefox ESR 24 (Bug 870540). On this index page, 24.1.x should be
-        # fallen under 24.0. This patter is a tricky part.
-        major_pattern = r'^' + \
-            re.escape(
-                ('%s' if major_version < 4 else '%g') % round(major_version, 1))
+        # fallen under 24.0. This pattern is a tricky part.
+        if int(major_version) in esr_major_versions:
```
Comment (pmclanahan) [inline bedrock/releasenotes/views.py]: A bit of repitition here. the previous version was a little crazy, but what about something like:

```python
converter = '%s' if int(major_version) in esr_major_versions else '%g'
major_pattern = r'^' + re.escape(converter % round(major_version, 1))
```

**Comment by pmclanahan** [comment]
I tested locally and this works well. I'd prefer there to be a test covering the new sorting, but it'd be nice to get this out if that will take too long. Hopefully a test of proper ordering wouldn't be too bad.

File: bedrock/releasenotes/views.py (hunk 1 above)
Comment (kyoshino) [inline bedrock/releasenotes/views.py]: Will use `converter` :)

**Comment by kyoshino** [comment]
I'll add tests.

**Comment by kyoshino** [comment]
Added tests to check the ordering.

**Comment by pmclanahan** [comment]
Looking good! I was just thinking how nice it would be to show some indication of release date on the releases page, maybe just for major version releases. Maybe I'll file a bug for that since I now realize we have that data.

**Pull request merged**